src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -659,6 +662,57 @@ static void moveTowards(IPlayerContext ctx, MovementState state, BlockPos pos) {
         )).setInput(Input.MOVE_FORWARD, true);
     }
 
+    static void moveTowardsWithoutRotation(IPlayerContext ctx, MovementState state, BlockPos dest) {
+        float ax = Mth.sin(ctx.playerRotations().getYaw() * DEG_TO_RAD_F);
+        float az = Mth.cos(ctx.playerRotations().getYaw() * DEG_TO_RAD_F);
+        Rotation blockRotation = RotationUtils.calcRotationFromVec3d(ctx.playerHead(),
+                VecUtils.getBlockPosCenter(dest),
+                ctx.playerRotations());
+        int selection = getSelection(blockRotation, ax, az);
+        switch (selection) {
+            case 0 -> state.setInput(Input.MOVE_FORWARD, true);
+            case 1 -> state.setInput(Input.MOVE_BACK, true);
+            case 2 -> state.setInput(Input.MOVE_LEFT, true);
+            case 3 -> state.setInput(Input.MOVE_RIGHT, true);
+            case 4 -> state.setInput(Input.MOVE_FORWARD, true).setInput(Input.MOVE_LEFT, true);
+            case 5 -> state.setInput(Input.MOVE_FORWARD, true).setInput(Input.MOVE_RIGHT, true);
+            case 6 -> state.setInput(Input.MOVE_BACK, true).setInput(Input.MOVE_LEFT, true);
+            case 7 -> state.setInput(Input.MOVE_BACK, true).setInput(Input.MOVE_RIGHT, true);
+            default -> {}
+        }
+    }
+
+    private static int getSelection(Rotation blockRotation, float ax, float az) {
+        float targetAx = Mth.sin(blockRotation.getYaw() * DEG_TO_RAD_F);
+        float targetAz = Mth.cos(blockRotation.getYaw() * DEG_TO_RAD_F);
+        Vec2[] options = getOptions(ax, az);
+        int selection = -1;
+        float closestX = 100000;
```

Comment:
I don't really like this, I feel like better would be to use streams. 

This function should look something like:
```
targetX = blah
targetZ = blah
getOptions(ax, az).stream().min(Comparator.comparingInt(option -> option.distanceTo(targetX, targetZ)).get()
```

Reply:
And your math is kinda weird too, really the comparison is "what yaw is closest to my desired yaw" but you're doing trig? I don't think it's necessary. I think it would be better to compare differences in yaw rotations rather than distances in (x,z)